sdk/lib/rtl/rtlp.h

```diff
@@ -27,10 +27,10 @@ extern VOID FASTCALL CHECK_PAGED_CODE_RTL(char *file, int line);
 #endif
 
 #define ROUND_DOWN(n, align) \
-    (((ULONG)(n)) & ~((align) - 1l))
+    (((ULONG_PTR)(n)) & ~((align) - 1l))
 
 #define ROUND_UP(n, align) \
-    ROUND_DOWN(((ULONG)(n)) + (align) - 1, (align))
+    ROUND_DOWN(((ULONG_PTR)(n)) + (align) - 1, (align))
```

Comment:
This must be carefully synced with the other ALIGN_MEM_xxx macros etc everywhere else (FreeLdr and elsewhere).
I don't know whether @tkreuzer or @ThFabba have some particular comments about that?

Reply:
Are there any ALIGM_MEM_...? Can't find them

Reply:
ok, these are called ALIGN_UP_... and ALIGN_DOWN_... :
https://git.reactos.org/?p=reactos.git&a=search&h=HEAD&st=grep&s=ALIGN_UP
https://git.reactos.org/?p=reactos.git&a=search&h=HEAD&st=grep&s=ALIGN_DOWN

Also I think some other ROUND_UP (not necessary in the kernel, but in other places in the code, but related to manipulating (what would be 64 bits) points) may need fixup:
https://git.reactos.org/?p=reactos.git&a=search&h=HEAD&st=grep&s=ROUND_UP
and check also for ROUND_DOWN ...